pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -95,6 +102,11 @@ func normalizeClusters(clusters []*v2.Cluster) []*v2.Cluster {
 func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environment, proxy *model.Proxy, push *model.PushContext,
 	services []*model.Service) []*v2.Cluster {
 	clusters := make([]*v2.Cluster, 0)
+	serviceAccounts, err := getServiceAccounts(env, proxy)
```

Comment:
Please stop making O(n ** 3) calls - any new code must pre-compute and use data in PushContext.

Reply:
[todo] add [ServiceAccounts to PushContext](https://github.com/istio/istio/blob/bb9944340ff8470df618b268f69f8390d1a5ff42/pilot/pkg/model/push_context.go#L67), otherwise it blocks merge from master to gcp-identity